src/main/java/tech/indicio/ariesmobileagentandroid/storage/Storage.java

```diff
@@ -12,39 +13,69 @@
 import java.util.concurrent.ExecutionException;
 
 import tech.indicio.ariesmobileagentandroid.IndyWallet;
+import tech.indicio.ariesmobileagentandroid.events.AriesEmitter;
+import tech.indicio.ariesmobileagentandroid.events.AriesEvent;
 
 public class Storage {
     private static final String TAG = "AMAA-Storage";
     private final IndyWallet indyWallet;
+    private final AriesEmitter eventEmitter;
 
     private final HashMap<String, Class<? extends BaseRecord>> recordClasses = new HashMap<>();
 
-    public Storage(IndyWallet indyWallet) {
-
+    public Storage(IndyWallet indyWallet, AriesEmitter eventEmitter) {
+        this.eventEmitter = eventEmitter;
         this.indyWallet = indyWallet;
     }
 
-    public void storeRecord(BaseRecord record) throws IndyException {
+    public void storeRecord(BaseRecord record) throws IndyException, ExecutionException, InterruptedException {
         Gson gson = new Gson();
         String type = record.getType();
         String id = record.id;
-        JSONObject tags = record.tags;
+        JsonObject tags = record.tags;
         String value = gson.toJson(record);
 
         //Logs for testing
         try {
-            String prettyString = new JSONObject(value).toString(4);
-            Log.d(TAG, "Storing record of type " + type + ".\n" + prettyString);
+            String prettyString = new JSONObject(value).toString(4).replaceAll("\\\\", "");
```

Comment:
These pretty strings could be a really good place to make a logging service, since that'd be more DRY and give additional options in the future. But that's a lil bit of a red herring anyways.